darwin/future/meta/objects/stage.py

```diff
@@ -75,6 +82,18 @@ def move_attached_files_to_stage(self, new_stage_id: UUID) -> Stage:
             self.meta_params["dataset_id"],
         )
         ids = [x.id for x in self.item_ids.collect_all()]
+
+        if wait:
+            while True:
```

Comment:
Adding retries makes total sense! Although, I couldn't get it working with [urllib3.util.Retry](https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html) as it dosen't have a `retry.wait()` - looks like it works best with `http.client`. For now, I've implemented something similar to this in a new commit.

Reply:
My take on this is that I like the NASA code rules approach:
> All loops must have fixed bounds. This prevents runaway code.

So, basically, I don't hate `while True` if the `while` block contains a max timeout or loop.

@Nathanjp91's approach is very sound, it could be recreated with a safe version using `while True`, but I'd want to see a maximum bound.